csharp/src/Ice/Properties.cs

```diff
@@ -518,6 +473,67 @@ public List<string> getUnusedProperties()
         }
     }
 
+    /// <summary>
+    /// Creates a new Properties object, loads the configuration files specified by the Ice.Config property or the
+    /// ICE_CONFIG environment variable, and parses the Ice command-line options.
+    /// </summary>
+    /// <param name="args">Command lines options.</param>
+    /// <param name="options">Options to configure the new Properties objects.</param>
+    /// <returns>A new Properties object.</returns>
+    internal static Properties loadConfigAndParseIceOptions(ref string[] args, PropertiesOptions options)
+    {
+        var properties = new Properties(options);
```

Comment:
All the logic from the previous "args parsing constructor" moved here without change.